features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/utils/Emoji.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2024 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.messages.impl.utils
+
+import com.sigpwned.emoji4j.core.Grapheme.Type.EMOJI
+import com.sigpwned.emoji4j.core.Grapheme.Type.PICTOGRAPHIC
+import com.sigpwned.emoji4j.core.GraphemeMatchResult
+import com.sigpwned.emoji4j.core.GraphemeMatcher
+
+private fun GraphemeMatchResult.isEmoji(): Boolean {
+    return grapheme().type == EMOJI || isEmojiPictographic()
+}
+
+private fun GraphemeMatchResult.isEmojiPictographic(): Boolean {
+    // Many non-"emoji" characters are pictographics. We only want to identify this specific range
+    // https://en.wikipedia.org/wiki/Miscellaneous_Symbols_and_Pictographs
+    return grapheme().type == PICTOGRAPHIC && group() in "🌍".."🗺"
+}
+
+/**
+ * Returns true if the string contains any "emoji or pictographic graphemes".
+ * @param max Upper limit of the number of emojis that should be in the string.
+ */
+fun String.containsEmojis(max: Int = Integer.MAX_VALUE): Boolean {
+    return GraphemeMatcher(this)
+        .results()
+        .filter(GraphemeMatchResult::isEmoji)
+        .count() in 1..max
+}
+
+/**
+ * Returns true if the string consists exclusively of "emoji or pictographic graphemes".
+ */
+fun String.containsOnlyEmojis(): Boolean {
+    val matcher = GraphemeMatcher(this)
+
+    var m: GraphemeMatchResult? = null
+    var contiguous = true
+    var prevend = 0
+    while (matcher.find()) {
+        m = matcher.toMatchResult()
+        contiguous = m!!.isEmoji() and contiguous and (m.start() == prevend)
+        prevend = m.end()
+    }
+
+    return isNotEmpty() and contiguous and (m?.end() == length)
```

Comment:
If the String is empty, it can return false at the beginning of this method, no need to instanciate a `GraphemeMatcher`.